packages/core/src/subscriber/chainSubscribers.ts

```diff
@@ -0,0 +1,170 @@
+import {
+    services,
+    utils,
+    config,
+    contracts,
+    database,
+} from '../../';
+import { ethers } from 'ethers';
+
+class ChainSubscribers {
+    provider: ethers.providers.WebSocketProvider;
+    ifaceCommunityAdmin: ethers.utils.Interface;
+    filterTopics: string[][];
+
+    constructor(provider: ethers.providers.WebSocketProvider) {
+        this.provider = provider;
+        this.ifaceCommunityAdmin = new ethers.utils.Interface(
+            contracts.CommunityAdminABI
+        );
+
+        this.filterTopics = [
+            [
+                ethers.utils.id(
+                    'CommunityAdded(address,address[],uint256,uint256,uint256,uint256,uint256,uint256,uint256)'
+                ),
+                ethers.utils.id('CommunityRemoved(address)'),
+            ],
+        ];
+        this.recover();
+    }
+
+    stop() {
+        this.provider.removeAllListeners();
+        services.app.ImMetadataService.setRecoverBlockUsingLastBlock();
+    }
+
+    recover() {
+        this._setupListener(this.provider);
+        // we start the listener alongside with the recover system
+        // so we know we don't lose events.
+        this._runRecoveryTxs(this.provider).then(() =>
+            services.app.ImMetadataService.removeRecoverBlock()
+        )
+    }
+
+    async _runRecoveryTxs(
+        provider: ethers.providers.WebSocketProvider
+    ) {
+        utils.Logger.info('Recovering past events...');
+        let startFromBlock: number;
+        let lastBlockCached = await database.redisClient.get('lastBlock');
+        if (!lastBlockCached) {
+            startFromBlock = await services.app.ImMetadataService.getRecoverBlock();
+        } else {
+            startFromBlock = parseInt(lastBlockCached)
+        }
+
+        const rawLogs = await provider.getLogs({
```

Comment:
Maybe I've not made myself clear before. In this provider.getLogs, we should use an http, so the provider should be a JSONRPC and not a websocket. Websockets are for real-time events. Here we are doing a one time request of past events.